openmls/src/group/public_group/validation.rs

```diff
@@ -308,6 +309,75 @@ impl PublicGroup {
         Ok(encryption_keys)
     }
 
+    /// Validate PreSharedKey proposals. This function implements the following checks:
+    /// // TODO: ValSem
+    /// * PSK in proposal must be of type Resumption/Application or External.
```

Comment:
This doesn't look right. Only resumption and external are PSK types. Application is a PSK usage. It looks like you're doing the right thing. But the documentation needs to updated to make clear what's what.

Reply:
I used `External` and `Resumption/<usage>` as shorthands but this is not very clear. Will update :+1: